docs/ingress-tls.md

```diff
@@ -0,0 +1,280 @@
+
+Convert the .crt certificate to pfx format and import it to Azure Key Vault. For example:
+
+```bash
+export AKV_NAME="[YOUR AKV NAME]"
+openssl pkcs12 -export -in ingress-tls.crt -inkey ingress-tls.key  -out $CERT_NAME.pfx
+# skip Password prompt
+
+az keyvault certificate import --vault-name $AKV_NAME -n $CERT_NAME -f $CERT_NAME.pfx
+```
+
+## Setup Cluster Prerequisites
+
+### Deploy Secrets-store CSI and the Azure Key Vault Provider
+
+[Follow this tutorial](https://github.com/Azure/secrets-store-csi-driver-provider-azure#install-the-secrets-store-csi-driver-and-the-azure-keyvault-provider) to setup the secret store csi driver and azure key vault provider in the cluster.
+
+### Optional: Deploy AAD Pod Identity
+
+If using AAD pod identity to access Azure Key Vault, make sure it is [configured properly](https://azure.github.io/aad-pod-identity/docs/demo/standard_walkthrough/) in the cluster.
+
+```bash
+export AAD_POD_IDENTITY_NAME=azure-kv
+```
+
+## Deploy a SecretsProviderClass Resource
+
+### Create a namespace
+
+```bash
+export NAMESPACE=ingress-test
+kubectl create ns $NAMESPACE
+```
+
+### Create the SecretProviderClass
+
+* To provide identity to access key vault, refer to the following [section](https://github.com/Azure/secrets-store-csi-driver-provider-azure#provide-identity-to-access-key-vault).
+* Set the `tenantId` and `keyvaultName`
+* If using **AAD pod identity** to access Azure Key Vault - set `usePodIdentity: "true"`
+* Use `objectType: secret` for the certificate, as this is the only way to retrieve the certificate and private key from azure key vault as documented [here](https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/master/docs/getting-certs-and-keys.md#getting-certificates-and-keys-using-azure-key-vault-provider)
+* Set secret type to `kubernetes.io/tls`
+
+```bash
+export TENANT_ID=[YOUR TENANT ID]
+
+cat <<EOF | kubectl apply -n $NAMESPACE -f -
+apiVersion: secrets-store.csi.x-k8s.io/v1alpha1
+kind: SecretProviderClass
+metadata:
+  name: azure-tls
+spec:
+  provider: azure
+  secretObjects:                                # secretObjects defines the desired state of synced K8s secret objects
+  - secretName: ingress-tls-csi
+    type: kubernetes.io/tls
+    data: 
+    - objectName: $CERT_NAME
+      key: tls.key
+    - objectName: $CERT_NAME
+      key: tls.crt
+  parameters:
+    usePodIdentity: "false"
+    keyvaultName: $AKV_NAME                        # the name of the KeyVault
+    objects: |
+      array:
+        - |
+          objectName: $CERT_NAME
+          objectType: secret
+    tenantId: $TENANT_ID                  # the tenant ID of the KeyVault
+EOF
+```
+
+## Deploy Ingress Controller
+
+### Add the official ingress chart repository
+
+```bash
+helm repo add ingress-nginx https://kubernetes.github.io/ingress-nginx
+helm repo update
+```
+
+### Deploy Ingress
+
+Depending on the TLS certificate lifecycle, follow one of the following steps:
```

Comment:
Thanks for the PR @balteravishay! Can you please add few comments around when a user would choose to add the CSI volume to the ingress controller deployment and when they would choose to add it an application deployment?

Reply:
Resolved as it has been address [here](https://github.com/Azure/secrets-store-csi-driver-provider-azure/pull/282/files#diff-2da0ddc89b1dfa84a2cff2521f42abfbafb87540a6bd531f2ca1cdd32ad3aed6R6-R9)